### ISSUE TITLE
Make parse_apicast_creds not mutating

### ISSUE THREAD

**Issue by mayorova** [3scale/apicast-xc#28]
The function `parse_apicast_creds` in [utils](https://github.com/3scale/apicast-xc/blob/master/xc/utils.lua#L10) is currently modifying the passed `creds` table.
I'd suggest to make it non-mutating, in case the original table needs to be reused somewhere in the code.

**Comment by davidor** [comment]
Agreed. It would be safer if that method did not mutate the table received in the parameters.
If XC is the only module used, it should be fine anyway, but it could be a problem when used in combination with other modules.
